src/detect-parse.c

```diff
@@ -1507,6 +1507,11 @@ int DetectSignatureSetAppProto(Signature *s, AppProto alproto)
         }
     }
 
+    if (AppLayerProtoDetectGetProtoName(alproto) == NULL && alproto != ALPROTO_HTTP) {
```

Comment:
why is there an exception for `ALPROTO_HTTP`?

Reply:
Because we can have rules with `alert http` but there is no such app-layer protocol : there is http1 and http2
So, we can improve this by testing that `alert http` rules need to have at least one of http1 or http2 enabled...
What do you think ?

Reply:
I don't think I understand. Shouldn't every `alproto` value lead to a non-null result for `AppLayerProtoDetectGetProtoName`?

Reply:
> Shouldn't every alproto value lead to a non-null result for AppLayerProtoDetectGetProtoName?`

Protocols disabled in the configuration lead to a NULL value cf `AppLayerProtoDetectRegisterProtocol`
There is `AppProtoToString` which returns a non NULL value for every alproto

Does that answer your question ?

Reply:
What kind of rule(+config?) case would hit this condition?

Reply:
Rules like
```
alert ip any any -> any any (enip_command: 112; sid:2; )
alert ip any any -> any any (dnp3_func: 1; sid:3; )
```

with default config (where DNP3 and ENIP are disabled)

Reply:
I meant the case where we the `alproto != ALPROTO_HTTP` condition.

Reply:
Could `AppLayerProtoDetectGetProtoName` return "http_any", "http1" or "http2" depending on which is enabled?

Reply:
Tried in https://github.com/OISF/suricata/pull/6888